plugins/org.python.pydev/tests_analysis/com/python/pydev/analysis/refactoring/quick_fixes/AddTokenAndImportStatementTest.java

```diff
@@ -410,6 +410,218 @@ public void testAllCases() throws Exception {
+                "    ActionResultDict,\\\r\n" +
+                "    UploadActivityParamsDict,\\\r\n" +
+                "    CloudLoginParamsDict,\\\r\n" +
+                "    ActionResult,\\  # some comment\r\n" +
+                "";
+        String expectedDoc = "from mod2 import ActivityInfoDict,\\\r\n" +
+                "    WorkspaceInfoDict,\\\r\n" +
+                "    PackageInfoDict,\\\r\n" +
+                "    ActionResultDict,\\\r\n" +
+                "    UploadActivityParamsDict,\\\r\n" +
+                "    CloudLoginParamsDict,\\\r\n" +
+                "    ActionResult, UploadNewActivityParamsDict,\\  # some comment\r\n" +
+                "";
+        checkLocalImport(baseDoc, "from mod2 import UploadNewActivityParamsDict", expectedDoc, true);
+    }
+
+    public void testGroupImport10() throws Exception {
+        String baseDoc = "from mod2 import (\r\n" +
+                "    ActivityInfoDict,\r\n" +
+                "    WorkspaceInfoDict,\r\n" +
+                "    PackageInfoDict,\r\n" +
+                "    ActionResultDict,\r\n" +
+                "    UploadActivityParamsDict,\r\n" +
+                "    CloudLoginParamsDict,\r\n" +
+                "    ActionResult) #some comment";
+        String expectedDoc = "from mod2 import (\r\n" +
+                "    ActivityInfoDict,\r\n" +
+                "    WorkspaceInfoDict,\r\n" +
+                "    PackageInfoDict,\r\n" +
+                "    ActionResultDict,\r\n" +
+                "    UploadActivityParamsDict,\r\n" +
+                "    CloudLoginParamsDict,\r\n" +
+                "    ActionResult, UploadNewActivityParamsDict) #some comment";
+        checkLocalImport(baseDoc, "from mod2 import UploadNewActivityParamsDict", expectedDoc, true);
+    }
+
+    public void testGroupImport11() throws Exception {
+        String baseDoc = "from mod2 import (\r\n" +
+                "    ActivityInfoDict,\r\n" +
+                "    WorkspaceInfoDict,\r\n" +
+                "    PackageInfoDict,\r\n" +
+                "    ActionResultDict,\r\n" +
+                "    UploadActivityParamsDict,\r\n" +
+                "    CloudLoginParamsDict,\r\n" +
+                "    ActionResult\r\n" +
+                "    ) #some comment";
+        String expectedDoc = "from mod2 import (\r\n" +
+                "    ActivityInfoDict,\r\n" +
+                "    WorkspaceInfoDict,\r\n" +
+                "    PackageInfoDict,\r\n" +
+                "    ActionResultDict,\r\n" +
+                "    UploadActivityParamsDict,\r\n" +
+                "    CloudLoginParamsDict,\r\n" +
+                "    ActionResult, UploadNewActivityParamsDict\r\n" +
+                "    ) #some comment";
+        checkLocalImport(baseDoc, "from mod2 import UploadNewActivityParamsDict", expectedDoc, true);
+    }
+
+    public void testGroupImport12() throws Exception {
+        String baseDoc = "from mod2 import (ActivityInfoDict #some comment\r\n" +
```

Comment:
This line of the test should be:

`"from mod2 import (ActivityInfoDict #some comment\\\r\n"`

Reply:
Sorry, I just saw that this was added earlier.